test/gateway.ts

```diff
@@ -11,23 +15,26 @@ import graphqlUploadExpress from 'graphql-upload/graphqlUploadExpress.js';
 import http from 'http';
 import type { AddressInfo } from 'net';
 
-import FileUploadDataSource from '../lib';
 import { ServiceDescription } from './gen-service';
+import { FileUploadDataSourceArgs } from '../lib/FileUploadDataSource';
 
 export type GatewayDescription = ServiceDescription & {
   gateway: ApolloGateway;
 };
 
-const gateway = async ({
-  chunkedAddress,
-  downloadAddress,
-}: {
-  chunkedAddress: AddressInfo;
-  downloadAddress: AddressInfo;
-}): Promise<GatewayDescription> => {
+const gateway = async <DS extends RemoteGraphQLDataSource>(
+  DataSource: new (config: FileUploadDataSourceArgs) => DS,
+  {
+    chunkedAddress,
+    downloadAddress,
+  }: {
+    chunkedAddress: AddressInfo;
+    downloadAddress: AddressInfo;
+  },
+): Promise<GatewayDescription> => {
   const apolloGateway = new ApolloGateway({
-    buildService: ({ url }): FileUploadDataSource =>
-      new FileUploadDataSource({
+    buildService: ({ url }): DS =>
+      new DataSource({
```

Comment:
Why use generics here? Couldn't you just do this?
```suggestion
    buildService: ({ url }): DataSourceWithCustomHeaders =>
      new DataSourceWithCustomHeaders({
```

Reply:
In the beginning, I was considering running some tests with  `DataSourceWithCustomHeaders` and others with just `FileUploadDataSource`. But the other tests are very independent, so I didn't need to bring two servers up. 

But I decided to keep it. Helps with debug. If the tester creates a different `RemoteDataSource` class and want to test it, he just have to change it here:
https://github.com/profusion/apollo-federation-file-upload/blob/90e307d91e76d2e57b0adabcc0340385774b5f24/test/index.ts#L55

Reply:
Bah, I'll use your suggestion. Not a big deal.